src/linear_samplers/block_col_FJLT.jl

```diff
@@ -0,0 +1,88 @@
+sampled.
+
+# Fields
+- `block_size::Int64`, the size of the sketching dimension
+- `sparsity::Float64`, the sparsity of the sampling matrix, should be between 0 and 1
+- `padded_size::Int64`, the size of the matrix when padded
+- `sampling_matrix::Union{SparseMatrixCSC, Nothing}`, storage for sparse sketching matrix 
+- `hadamard::Union{AbstractMatrix, Nothing}`, storage for the hadamard matrix.
+- `Ap::Union{AbstractMatrix, Nothing}`, storage for padded matrix
+- `bp::Union{AbstractMatrix, Nothing}`, storage for padded vector
+- `signs::Union{Vector{Bool}, Nothing}`, storage for random sign flips.
+- `scaling::Float64`, storage for the scaling of the sketches.
+
+Calling `LinSysBlockColFJLT()` defaults to setting `sparsity` to .3 and the blocksize to 2.
+
+Nir Ailon and Bernard Chazelle. 2006. Approximate nearest neighbors and the fast Johnson-Lindenstrauss transform. In Proceedings of the thirty-eighth annual ACM symposium on Theory of Computing (STOC '06). Association for Computing Machinery, New York, NY, USA, 557–563. https://doi.org/10.1145/1132516.1132597
+"""
+mutable struct LinSysBlockColFJLT <: LinSysBlkColSampler
+    block_size::Int64
+    sparsity::Float64 
+    padded_size::Int64
+    sampling_matrix::Union{SparseMatrixCSC, Nothing}
+    hadamard::Union{AbstractMatrix, Nothing}
+    Ap::Union{AbstractMatrix, Nothing}
+    bp::Union{AbstractVector, Nothing}
+    signs::Union{Vector{Bool}, Nothing}
+    scaling::Float64
+end
+
+LinSysBlockColFJLT(;blocksize = 2, sparsity = .3) = LinSysBlockColFJLT(
+                                                   blocksize,
+                                                   sparsity, 
+                                                   0, 
+                                                   nothing, 
+                                                   nothing,
+                                                   nothing,
+                                                   nothing,
+                                                   nothing,
+                                                   0.0
+                                                  )
+
+# Common sample interface for linear systems
+function sample(
+    type::LinSysBlockColFJLT,
+    A::AbstractArray,
+    b::AbstractVector,
+    x::AbstractVector,
+    iter::Int64
+)
+    if iter == 1
+        m, n = size(A)
+        # If matrix is not a power of 2 then pad the rows
+        if rem(log(2, n), 1) != 0
+            type.padded_size = Int64(2^(div(log(2, n), 1) + 1)) 
+            # Find nearest power 2 and allocate
+            type.Ap = zeros(m, type.padded_size)
+            # Pad matrix and constant vector
+            type.Ap[:, 1:n] .= A
+        else
+            type.padded_size = n
+            type.Ap = A
+        end
+        type.hadamard = hadamard(type.padded_size)
+        # Compute scaling and sign flips
+        type.scaling = sqrt(type.block_size / (type.padded_size * type.sparsity))
```

Comment:
Where is this scaling coming from? I do not see this in the reference. Presumably, you are trying to scale by `1/sqrt(d*q)` where Ailon & Chazelle define `q = min( log(n)^2 / log(d) , 1)` assuming `p=2`. Or is this coming from somewhere else?

Reply:
Yes, that was the scaling I was attempting to do. I had gotten the SRHT scalings and this one confused.